test/e2e/basic/simple-kmod.go

```diff
@@ -2,223 +2,201 @@ package e2e
-
+	return framework.CreateFromYAML(simpleKmodCrYAML, cl)
 }
 
-func simpleKmodCreate(cs *framework.ClientSet, cl client.Client) {
-
-	var err error
-
-	simpleKmodCrYAML, err = ioutil.ReadFile("../../../charts/example/simple-kmod-0.0.1/simple-kmod.yaml")
+func deleteSimpleKmod(cl client.Client) error {
+	simpleKmodCrYAML, err := ioutil.ReadFile(simpleKmodChartPath)
 	if err != nil {
-		panic(err)
+		return err
 	}
-	framework.CreateFromYAML(simpleKmodCrYAML, cl)
-
-	/*
-		ginkgo.By("waiting for completion driver-container-base build")
-		err = wait.PollImmediate(pollInterval, waitDuration, func() (bool, error) {
-
-			dcbPods, err := cs.Pods("driver-container-base").List(context.TODO(), metav1.ListOptions{})
-			if err != nil {
-				return false, fmt.Errorf("Error getting list of pods, %v", err)
-			}
+	return framework.DeleteFromYAMLWithCR(simpleKmodCrYAML, cl)
+}
 
-			if len(dcbPods.Items) < 1 {
-				return false, nil
+func waitImageReady(clientSet *kubernetes.Clientset) error {
+	watch, err := clientSet.CoreV1().Pods(simpleKmodName).Watch(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	quitChan := make(chan struct{})
+	go func() {
+		for event := range watch.ResultChan() {
+			p, ok := event.Object.(*corev1.Pod)
+			if !ok {
+				continue
 			}
-
-			var sum = len(dcbPods.Items)
-			for _, pod := range dcbPods.Items {
-				// Get logs of driverContainerBase
-				_ = GetRecentPodLogs(pod.GetName(), pod.GetNamespace(), pollInterval)
-				if pod.Status.Phase == "Succeeded" {
-					sum = sum - 1
-				}
+			if _, ok := p.GetAnnotations()["openshift.io/build.name"]; !ok {
+				continue
 			}
-
-			if sum == 0 {
-				return true, nil
+			if p.Status.Phase == corev1.PodSucceeded {
+				quitChan <- struct{}{}
+				return
 			}
-
-			return false, nil
-		})
-		if err != nil {
-			explain = err.Error()
-		}
-		gomega.Expect(err).NotTo(gomega.HaveOccurred(), explain)
-	*/
-
-	var dss *v1.DaemonSetList
-
-	for {
-		time.Sleep(10 * time.Second)
-
-		ginkgo.By("Waiting for simple-kmod daemonset to be ready")
-		dss, err = cs.DaemonSets("simple-kmod").List(context.TODO(), metav1.ListOptions{})
-		if err != nil {
-			_, _ = Logf("Looking for DaemonSet", err)
-			gomega.Expect(err).NotTo(gomega.HaveOccurred(), explain)
 		}
+	}()
+	<-quitChan
+	return nil
+}
 
-		if len(dss.Items) > 0 {
-			break
-		}
+func deleteDaemonSetPods(clientSet *kubernetes.Clientset) error {
+	// dont wait for backoff
+	list, err := clientSet.CoreV1().Pods(simpleKmodName).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return err
 	}
-
-	for _, ds := range dss.Items {
-		err = WaitForDaemonsetReady(cs, pollInterval, waitDuration, "simple-kmod", ds.Name)
-		if err != nil {
-			explain = err.Error()
+	for _, pod := range list.Items {
+		if pod.Status.Phase == corev1.PodFailed {
+			err = clientSet.CoreV1().Pods(simpleKmodName).Delete(context.Background(), pod.Name, metav1.DeleteOptions{})
```

Comment:
This is to avoid bad timing with the backoff. In order to shave some time from the test we delete pod intentionally before Kubernetes, so its a tad faster.

Reply:
Could you please add a comment explaining this ?

Reply:
There was already a comment, but I extended it. Let me know what you think.